src/database.rs

```diff
@@ -93,26 +98,26 @@ impl DatabaseRef for ReadOnlyDB {
 
     /// Get block hash by block number.
     fn block_hash_ref(&self, _: u64) -> Result<B256, Self::Error> {
-        unimplemented!("BLOCKHASH is disabled")
```

Comment:
We are actually not implementing this method, so `unimplemented!` is better I think.

Reply:
If the plan is to implement it, then I will revert this change